oci/spec_opts.go

```diff
@@ -824,6 +824,7 @@ func WithAddedCapabilities(caps []string) SpecOpts {
 				&s.Process.Capabilities.Effective,
 				&s.Process.Capabilities.Permitted,
 				&s.Process.Capabilities.Inheritable,
+				&s.Process.Capabilities.Ambient,
```

Comment:
This code change is so simple. I wonder how you achieve that defaultUnixCaps isn't added?

Reply:
They are explicitly added to Inheritable, Effective, Permitted and Bounding sets here: https://github.com/containerd/containerd/blob/f0a32c66dad1e9de716c9960af806105d691cd78/oci/spec.go#L132

Reply:
note from the cri config doc:
```
      # base_runtime_spec is a file path to a JSON file with the OCI spec that will be used as the base spec that all
      # container's are created from.
      # Use containerd's `ctr oci spec > /etc/containerd/cri-base.json` to output initial spec file.
      # Spec files are loaded at launch, so containerd daemon must be restarted on any changes to refresh default specs.
      # Still running containers and restarted containers will still be using the original spec from which that container was created.
      base_runtime_spec = ""
```

Reply:
while you can add ambient in your specified cri-base.json... we explicitly nil it out currently..

Reply:
Which I think is the right thing to do. If we added defaultUnixCaps to ambient set then any non-root container would have all the default privileges and that would be very very bad!